c2corg_ui/static/js/apiservice.js

```diff
@@ -391,6 +391,20 @@ app.Api.prototype.login = function(data) {
 
 
 /**
+ * @param {Object} data
+ * @return {!angular.$q.Promise<!angular.$http.Response>}
+ */
+app.Api.prototype.ssoLogin = function(data) {
+  var promise = this.postJson_('/sso_login', data);
+  promise.catch(function(response) {
+    var msg = this.alerts_.gettext('Login failed:');
```

Comment:
Perhaps you could use a more specific error message (``SSO login failed:``?) to make it more obvious if something wrong happens?